c++/src/kj/refcount.h

```diff
@@ -115,6 +142,108 @@ Own<T> Refcounted::addRefInternal(T* object) {
   return Own<T>(object, *refcounted);
 }
 
+template <typename T>
+Rc<T> Refcounted::addRcRefInternal(T* object) {
+  Refcounted* refcounted = object;
+  ++refcounted->refcount;
+  return Rc<T>(object);
+}
+
+template<typename T>
+class Rc {
+  // Smart pointer for reference counted objects. 
+  //
+  // There are only two ways to obtain new Rc instances:
+  // - use kj::rc<T>(...) function to create new T.
+  // - use kj::Rc::addRef() and the existing Rc instance.
+  // - use EnableAddRefToThis to allow T instance to add new references to itself.
+  //
+  // kj::Rc<T> can be used everywhere Own<T> is expected.
```

Comment:
This line still seems incorrect -- you have to explicitly use `toOwn()` to convert it.

Reply:
argh, I deleted the wrong line (next sentence instead).
Fixed and rewrote into a separate `-` item. Sorry for that.